kolena/_experimental/dataset/_evaluation.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from dataclasses import asdict
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterator
+from typing import List
+from typing import Optional
+
+try:
+    from typing import Protocol
+except ImportError:
+    from typing_extensions import Protocol
+try:
+    from typing import runtime_checkable
+except ImportError:
+    from typing_extensions import runtime_checkable
+from typing import Tuple
+from typing import Union
+
+import pandas as pd
+
+from kolena._api.v2.evaluation import LoadMetricsRequest
+from kolena._api.v2.evaluation import Path as EvaluationPath
+from kolena._api.v2.evaluation import UploadMetricsRequest
+from kolena._api.v2.model import LoadInferencesRequest
+from kolena._api.v2.model import Path
+from kolena._api.v2.model import UploadInferencesRequest
+from kolena._experimental.dataset._dataset import _iter_dataset_raw
+from kolena._experimental.dataset._dataset import _to_deserialized_dataframe
+from kolena._experimental.dataset._dataset import _to_serialized_dataframe
+from kolena._experimental.dataset.common import COL_DATAPOINT
+from kolena._experimental.dataset.common import COL_EVAL_CONFIG
+from kolena._experimental.dataset.common import COL_INFERENCE
+from kolena._experimental.dataset.common import COL_METRICS
+from kolena._experimental.dataset.common import validate_batch_size
+from kolena._utils import krequests_v2 as krequests
+from kolena._utils import log
+from kolena._utils.batched_load import _BatchedLoader
+from kolena._utils.batched_load import init_upload
+from kolena._utils.batched_load import upload_data_frame
+from kolena._utils.consts import BatchSize
+from kolena._utils.state import API_V2
+from kolena.errors import IncorrectUsageError
+
+
+@runtime_checkable
+class EvaluationConfiguration(Protocol):
+    def _to_dict(self) -> Dict[str, Any]:
+        ...
+
+
+TYPE_EVALUATION_CONFIG = Union[EvaluationConfiguration, Dict]
```

Comment:
`Dict[str, Any]`?

Reply:
just adapted the freeform cfg at [here](https://github.com/kolenaIO/kolena/pull/252/commits/a1cf901f3d2f9fc42098acca550c61179a04b3c2)

Reply:
@gordonhart did you mean to only support `Dict[str, Any]`? I initally added `EvaluationConfiguration` as duck-typing to support if users want to use `class`, particularly existing `EvaluatorConfiguration`.

Reply:
just for the ctx, i followed [this discussion](https://linear.app/kolena/issue/KOL-3422/update-eval-config-concept-for-datasets#comment-ac77b6e7) to update